src/js/providers/bandwidth-monitor.ts

```diff
@@ -1,13 +1,20 @@
 import { BANDWIDTH_ESTIMATE } from 'events/events';
 import { isValidNumber } from 'utils/underscore';
+import { DefaultProvider } from 'types/generic.type';
 
-export default function BandwidthMonitor(provider, initialEstimate) {
-    let bandwidthMonitorInterval = null;
+export type BandwidthMonitor = {
+    start: () => void;
+    stop: () => void;
+    getEstimate: () => number;
+}
+
+export default function BandwidthMonitor(provider: DefaultProvider, initialEstimate: number): BandwidthMonitor {
+    let bandwidthMonitorInterval: number | undefined;
     let bandwidthEstimate = initialEstimate;
     return {
-        start() {
+        start(): void {
             this.stop();
-            setInterval(() => {
+            bandwidthMonitorInterval = window.setInterval(() => {
```

Comment:
It's defaulting to using NodeJS typings so this was needed to explicitly use the window.setInterval

Reply:
We do the same thing in hls.js (`window.setInterval` or `self.setInterval` and same for `setTimeout`).